### PR TITLE
Opening .sqlite and .charts with Strikelines and coping them to Osmand through AIDL

### PR DESCRIPTION
**Pull request by MadWasp79**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -45,6 +44,25 @@
         <data android:scheme="strike_lines_app"/>
       </intent-filter>
```
Comment (vshcherb) [inline app/src/main/AndroidManifest.xml:46]: we need to have .charts extension support as well, though it should be renamed while copying to OsmAnd

Comment (MadWasp79) [reply app/src/main/AndroidManifest.xml:46]: fixed

File: app/src/main/java/com/strikelines/app/ImportHelper.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                if (response) {
+                    receivedData += read
+                    read = bis.read(data, 0, data.size)
+                    sentData += read
+                    counter++
+                }
+
+                response = app.osmandHelper.copyFile(CopyFileParams(fileName, fileSize, receivedData, data, copyStartTime))
+
+            } while (read != 0)
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ImportHelper.kt]: Deadlock

File: app/src/main/java/com/strikelines/app/ui/MainActivity.kt [hunk 3]
```diff
@@ -125,15 +131,60 @@ class MainActivity : AppCompatActivity(), OsmandHelperListener {
...
-		osmandHelper.onOsmandInitCallback = osmandHelperInitListener
+		osmandHelper.onOsmandInitCallbacks.add(osmandHelperInitListener)
+		val intent:Intent? = intent
+			if (Intent.ACTION_VIEW == intent?.action) {
+				if (intent.data != null) {
+					val data = intent.data
+					val scheme = data!!.scheme
+					intent.action = null
+					if (scheme == "content") {
+						var uri = "file:///storage/emulated/0/" + data.toString().substringAfter("external_files/")
```
Comment (crimean) [inline app/src/main/java/com/strikelines/app/ui/MainActivity.kt]: Hardcode

File: app/src/main/java/com/strikelines/app/ImportHelper.kt (hunk 2 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/ImportHelper.kt]: fixed

**Pull request merged**